genesyscloud/resource_genesyscloud_integration.go

```diff
@@ -0,0 +1,495 @@
+		if len(typeInput) > 0 {
+			inputMap := typeInput[0].(map[string]interface{})
+			// Only set non-empty values.
+			if id := inputMap["id"].(string); len(id) > 0 {
+				inteType.Id = &id
+			}
+			if name := inputMap["name"].(string); len(name) > 0 {
+				inteType.Name = &name
+			}
+			if description := inputMap["description"].(string); len(description) > 0 {
+				inteType.Description = &description
+			}
+			if provider := inputMap["provider"].(string); len(provider) > 0 {
+				inteType.Provider = &provider
+			}
+			if category := inputMap["category"].(string); len(category) > 0 {
+				inteType.Category = &category
+			}
+		}
+		return &inteType
+	}
+	return nil
+}
+
+func updateIntegrationConfig(d *schema.ResourceData, integrationAPI *platformclientv2.IntegrationsApi) diag.Diagnostics {
+	if d.HasChange("config") {
+		if configInput := d.Get("config").([]interface{}); configInput != nil {
+			var config platformclientv2.Integrationconfiguration
+			integrationConfig, _, err := integrationAPI.GetIntegrationConfigCurrent(d.Id())
+
+			if err != nil {
+				diag.Errorf("Failed to get the integration config for integration %s before updating its config: %s", d.Id(), err)
+			}
+			if len(configInput) > 0 {
+				configMap := configInput[0].(map[string]interface{})
+
+				name := configMap["name"].(string)
+				config.Name = &name
+
+				config.Version = integrationConfig.Version
+
+				notes := configMap["notes"].(string)
+				config.Notes = &notes
+
+				propJSON := make(map[string]interface{})
+				if properties := configMap["properties"].(string); len(properties) > 0 {
+					if err := json.Unmarshal([]byte(properties), &propJSON); err != nil {
+						return diag.Errorf("Failed to convert properties string to JSON for integration %s: %s", d.Id(), err)
+					}
+				}
+				config.Properties = &propJSON
+
+				advJSON := make(map[string]interface{})
+				if advanced := configMap["advanced"].(string); len(advanced) > 0 {
+					if err := json.Unmarshal([]byte(advanced), &advJSON); err != nil {
+						return diag.Errorf("Failed to convert advanced property string to JSON for integration %s: %s", d.Id(), err)
+					}
+				}
+				config.Advanced = &advJSON
+
+				credentialMap := make(map[string]platformclientv2.Credentialinfo)
+				if credentials := configMap["credentials"].(string); len(credentials) > 0 {
+					credJSON := make(map[string]interface{})
+					if err := json.Unmarshal([]byte(credentials), &credJSON); err != nil {
+						return diag.Errorf("Failed to convert credentials property string to JSON for integration %s: %s", d.Id(), err)
+					}
+					for k, v := range credJSON {
+						for _, value := range v.(map[string]interface{}) {
+							credentialId := value.(string)
+							credentialMap[k] = platformclientv2.Credentialinfo{Id: &credentialId}
+						}
+					}
+				}
+				config.Credentials = &credentialMap
+
+			}
+
+			diagErr := retryWhen(isVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
```

Comment:
Same as above. I wouldn't expect us to need to handle version mismatch errors for the integration APIs.